### PR TITLE
Kubebuilder Annotation Document

### PR DESCRIPTION
**Pull request by fanzhangio**
- Syntax of annotation
- Symbols allowed in annotation

### CODE REVIEW

**Comment by fanzhangio** [comment]
#554

File: docs/kubebuilder_annotation.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  Comma `,` is the 2nd level delimiter (to annotation) for splitting key-value pairs in **key-value elements** which is normally the last token in annotation. e.g. `+kubebuilder:printcolumn:name=<name>,type=<type>,description=<desc>,JSONPath:<.spec.Name>,priority=<int32>,format=<format>` It works within token which is the 2nd level of annotation, so it is called "2nd level delimiter"
+
+- **Equal sign**
+
+  Equal sign `=` is the 3rd level delimiter (to annotation) for identifying key and value. Since the `key=value` parts are splitted from single token (2nd level), its inner delimiter `=` works for next level (3rd level)
+
+- **Pipe sign or Vertical bar**
+
+  Pipe sign `|` is the 4th level delimiter, which works inside the `key=value` part (3rd level) indicating key and value.
+
```
Comment (droot) [inline docs/kubebuilder_annotation.md]: We use `;` also as delimiter ? I remember seeing it somewhere. Also pl. check with core k8s convention.

File: docs/kubebuilder_annotation.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Kubebuilder Annotation
+
+This document illustrates how the `Annotation` works in kubebuilder and how to use `Annotation` of kubebuilder.
```
[the remark this answers is not in the document]
Comment (fanzhangio) [reply docs/kubebuilder_annotation.md]: Awesome! I should merge motivation and introduction into one summary.

File: docs/kubebuilder_annotation.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Motivation
+
+`Kubebuilder Annotation` is designed for this kind of use case: To use kubebuilder, all you have to do is focus on writing your code, and put instructions with parameters as annotations along with your code, so that everything will be handled based on these annotations instructions by kubebuilder.
+
```
[the remark this answers is not in the document]
Comment (fanzhangio) [reply docs/kubebuilder_annotation.md]: Agree, the information can be addressed in introduction paragraph.

File: docs/kubebuilder_annotation.md (hunk 1 above)
Comment (fanzhangio) [reply docs/kubebuilder_annotation.md]: Right, `semicolon` is one of the delimiter. I missed this section in editing MD file. Added back.

File: docs/kubebuilder_annotation.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Kubebuilder Annotation applies `Modularized Annotation Pattern` and follows its syntax. The modularized annotation pattern is a general solution for a commonly occurring issue in developing and using Go tools and SDKs: developing and maintaining a tool or SDK of an idea workflow is difficult to coordinate, especially for an open source to expend or iterate its functionality. Code corruption will easily ruin a good project. Using command line tools normally requires inputting one or more sub commands and passing a series of flags and parameters, which is tedious and increases user’s burden.
+
```
[the remark this answers is not in the document]
Comment (fanzhangio) [reply docs/kubebuilder_annotation.md]: This is background information addressing the high level of the benefits to applying annotation. I hope users can understand the benefits why we use it in kubebuilder. WDYT ?

**Pull request merged**